txnsync/encodedgroupstypes.go

```diff
@@ -35,7 +35,7 @@ const maxAddressBytes = maxEncodedTransactionGroupEntries * crypto.DigestSize
 var errInvalidTxType = errors.New("invalid txtype")
 
 //msgp:allocbound txnGroups maxEncodedTransactionGroupEntries
-type txnGroups []transactions.SignedTxn
+type txnGroups transactions.SignedTxnSlice
```

Comment:
How will the allocbound work with the above `//msgp:allocbound txnGroups maxEncodedTransactionGroupEntries`?

Reply:
When creating the msgp decoder, it makes sure that the decoded buffer has less(or equal) number of entries to the one specified here. It's equivalent to using struct tags, which cannot be used here ( since it's not a struct.. )

Reply:
I see that we have an alloc bound here as well as for declaring the `transactions.SignedTxnSlice`, will it just apply both bounds?

Reply:
If you'll look on the generated code, you'll see that the decoder would now perform the `txnGroups` decoding without any bounds checking by deferring the decoding to the `transactions.SignedTxnSlice` decoder ( that does the bounds checking )